product/gradle-plugin/src/main/kotlin/PythonTasks.kt

```diff
@@ -338,7 +338,9 @@ internal class TaskBuilder(
         // Exclude .py files which have a corresponding .pyc, unless unless they’re
         // included in extractPackages.
         val excludePy = { fte: FileTreeElement ->
-            if (fte.name.endsWith(".py") &&
+            if ("*" in python.extractPackages) {
+                false
```

Comment:
This could be merged into the `any` block below, e.g. `it == "*" || fte.path.replace ...`